qualtran/bloqs/select_swap_qrom.py

```diff
@@ -151,7 +151,7 @@ def selection_registers(self) -> Tuple[SelectionRegister, ...]:
     @cached_property
     def target_registers(self) -> Tuple[Register, ...]:
         return tuple(
-            Register(f'target{sequence_id}', self._target_bitsizes[sequence_id])
+            Register(f'target{sequence_id}_', self._target_bitsizes[sequence_id])
```

Comment:
add a comment explaining why the underscore is necessary and link to issue

Reply:
Done.